### DISCUSSION TITLE
Add available CLI options for queued jobs in code

### DISCUSSION THREAD

**Discussion by therecluse26** [laravel/framework#47544] [Ideas]
Hi there! Currently, there appears to be no way to pass through options like timeouts and max attempts to queued jobs in code. Normally, in an environment where every job is written by me, I could just add a `$timeout` property to the job class, but with jobs from external packages, all I can do is request that they add a timeout to the job class (and this timeout likely wouldn't apply in the same way to every user. so there would have to be some negotiation).

Instead, I think it'd be very helpful to add the ability to manually pass these options through to jobs at runtime via things like the `dispatch()` helper, the `Bus::chain` method, etc. That way, users can tweak package job settings at-will in code.

A few possible ideas for how this could be implemented:
![image](https://github.com/laravel/framework/assets/8239106/03b83af1-dea7-40b3-b7cd-b5f288f9d0fd)

**Comment by henzeb** [comment]
Aren't said options already available?

Also, you can extend job classes and add those parameters there?

**Reply by therecluse26** [reply]
Are they? After some cursory digging, it doesn't appear so, otherwise I wouldn't have brought the idea forward.

And yes, they could just extend the class, but if that's a blocker to implementing this, what's the point of having the option available in the CLI?
